src/transactionsFactories/tokenManagementTransactionsFactory.ts

```diff
@@ -59,21 +61,25 @@ export class TokenManagementTransactionsFactory {
 
         const dataParts = [
             "issue",
-            utf8ToHex(options.tokenName),
-            utf8ToHex(options.tokenTicker),
-            numberToPaddedHex(options.initialSupply),
-            numberToPaddedHex(options.numDecimals),
-            utf8ToHex("canFreeze"),
+            ...this.argSerializer.valuesToStrings([
+                new StringValue(options.tokenName),
+                new StringValue(options.tokenTicker),
+            ]),
+            ...this.argSerializer.valuesToStrings([
+                new BigUIntValue(options.initialSupply),
+                new BigUIntValue(options.numDecimals),
```

Comment:
Can be merged into the call of `valuesToStrings()` above.

Reply:
Indeed, done!